openshift/mlflow-experiment-image-buildconfig.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: v1
+kind: Template
+
+labels:
+  application: ${APPLICATION_NAME}
+
+metadata:
+  name: mlflow-experiment-image-build-template
+
+parameters:
+- description: Application name
+  value: 'mlflow-experiment-image'
+  name: APPLICATION_NAME
+  required: true
+- name: GIT_URI
+  value: https://github.com/4n4nd/experiment-tracking-template.git
+  required: true
+
+
+objects:
+- apiVersion: image.openshift.io/v1
+  kind: ImageStream
+  metadata:
+    name: ${APPLICATION_NAME}
+    labels:
+      app: ${APPLICATION_NAME}
+  spec:
+    dockerImageRepository: ${APPLICATION_NAME}
+    tags:
+    - name: latest
+    lookupPolicy:
+      local: true
+
+- apiVersion: build.openshift.io/v1
+  kind: BuildConfig
+  metadata:
+    name: ${APPLICATION_NAME}
+    labels:
+      app: ${APPLICATION_NAME}
+  spec:
+    resources:
+      limits:
+        memory: 1Gi
+    output:
+      to:
+        kind: ImageStreamTag
+        name: ${APPLICATION_NAME}:latest
```

Comment:
Let's remove `:latest` and as users may want to use different tags

Reply:
@zmhassan if not `:latest`, what should I use as the tag?

Reply:
Leave it blank. Perhaps have another parameter called `IMAGE_URL`.

Reply:
In the build-config template? or the job template?

Reply:
So the purpose of the build-config i thought was to build the image. Then once you have a ready image you could run it in a job template with different parameters.

Reply:
I will make the image being used in the job-template configurable then